jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -91,7 +91,11 @@ def calculate_moment(self, add_data=True):
             Whether to add the resulting data object to the app according to ``add_results``.
         """
         # Retrieve the data cube and slice out desired region, if specified
-        cube = self.dataset.get_object(cls=Spectrum1D, statistic=None)
+        if "_orig_spec" in self.dataset.selected_obj.meta:
+            cube = self.dataset.selected_obj.meta["_orig_spec"]
+        else:
+            cube = self.dataset.selected_obj
```

Comment:
This grabs the original data from collection and bypasses going through glue-astronomy translator. Should work unless there is something I overlooked? Maybe try it with a few different cubes and see.